src/hooks/useBasePopoverReactionList/index.ts

```diff
@@ -0,0 +1,131 @@
+     */
+    function getReactionListMeasuredLocation(): Promise<{x: number; y: number}> {
+        return new Promise((resolve) => {
+            const reactionListAnchor = reactionListRef.current;
+            if (reactionListAnchor && 'measureInWindow' in reactionListAnchor) {
+                reactionListAnchor.measureInWindow((x, y) => resolve({x, y}));
+            } else {
+                // If the anchor is not available or does not have the measureInWindow method, we return 0, 0
+                resolve({x: 0, y: 0});
+            }
+        });
+    }
+
+    /**
+     * Show the ReactionList modal popover.
+     *
+     * @param event - Object -  A press event.
+     * @param reactionListAnchor - Element - reactionListAnchor
+     */
+    const showReactionList: ShowReactionList = (event, reactionListAnchor) => {
+        // We get the cursor coordinates and the reactionListAnchor coordinates to calculate the popover position
+        const nativeEvent = (event as SyntheticEvent<ReactionListAnchor, MouseEvent>)?.nativeEvent || {};
+        reactionListRef.current = reactionListAnchor;
+        getReactionListMeasuredLocation().then(({x, y}) => {
+            setCursorRelativePosition({horizontal: nativeEvent.pageX - x, vertical: nativeEvent.pageY - y});
+            setPopoverAnchorPosition({
+                horizontal: nativeEvent.pageX,
+                vertical: nativeEvent.pageY,
+            });
+            setIsPopoverVisible(true);
+        });
+    };
+
+    /**
+     * Hide the ReactionList modal popover.
+     */
+    function hideReactionList() {
+        setIsPopoverVisible(false);
+    }
+
+    useEffect(() => {
+        const dimensionsEventListener = Dimensions.addEventListener('change', () => {
+            if (!isPopoverVisible) {
+                // If the popover is not visible, we don't need to update the component
+                return;
+            }
+            getReactionListMeasuredLocation().then(({x, y}) => {
+                if (!x || !y) {
+                    return;
+                }
+                setPopoverAnchorPosition({
+                    horizontal: cursorRelativePosition.horizontal + x,
+                    vertical: cursorRelativePosition.vertical + y,
+                });
+            });
+        });
+
+        return () => {
+            dimensionsEventListener.remove();
+        };
+    }, [
+        isPopoverVisible,
+        reportActionID,
+        preferredLocale,
+        cursorRelativePosition.horizontal,
+        cursorRelativePosition.vertical,
+        popoverAnchorPosition.horizontal,
+        popoverAnchorPosition.vertical,
+    ]);
+
+    useEffect(() => {
+        if (!isPopoverVisible) {
+            // If the popover is not visible, we don't need to update the component
+            return;
+        }
+
+        // Hide the list when all reactions are removed
+        const users = emojiReactions?.[emojiName].users;
```

Comment:
```suggestion
        const users = emojiReactions?.[emojiName]?.users;
```
crash right clicking on reaction list

![Screenshot 2024-02-06 at 10 01 37 PM](https://github.com/Expensify/App/assets/75031127/775c20ff-0f44-45d3-9d91-fe07913d1c8f)

Reply:
happens rarely.